Sources/AblyChat/Rooms.swift

```diff
@@ -1,5 +1,42 @@
+@preconcurrency import Ably
+
 public protocol Rooms: AnyObject, Sendable {
-    func get(roomID: String, options: RoomOptions) throws -> any Room
+    func get(roomID: String, options: RoomOptions) async throws -> any Room
     func release(roomID: String) async throws
     var clientOptions: ClientOptions { get }
 }
+
+internal actor DefaultRooms: Rooms {
+    /// Exposed so that we can test it.
+    internal nonisolated let realtime: ARTRealtimeProtocol
+    internal nonisolated let clientOptions: ClientOptions
+
+    /// The set of rooms, keyed by room ID.
+    private var rooms: [String: DefaultRoom] = [:]
+
+    internal init(realtime: ARTRealtimeProtocol, clientOptions: ClientOptions) {
+        self.realtime = realtime
+        self.clientOptions = clientOptions
+    }
+
+    internal func get(roomID: String, options: RoomOptions) throws -> any Room {
+        // CHA-RC1b
+        if let existingRoom = rooms[roomID] {
+            if existingRoom.options != options {
+                throw ARTErrorInfo(
```

Comment:
Shouldn't options be just ignored if the room with the given id already exists?

Reply:
I don't know, I’d suggest commenting on the spec PR if you disagree with it.

Reply:
Left a [question](https://github.com/ably/specification/pull/200/files#r1736827643). Will resolve it here for now.